### DISCUSSION TITLE
Why are menu labels titlecased?

### DISCUSSION THREAD

**Discussion by SebCorbin** [wagtail/wagtail#12356] [Development]
In SnippetViewSet
https://github.com/wagtail/wagtail/blob/d526612e4befbb1bb80addb596830b38ab0dabde/wagtail/snippets/views/snippets.py#L824-L834

and in ModelViewSet
https://github.com/wagtail/wagtail/blob/d526612e4befbb1bb80addb596830b38ab0dabde/wagtail/admin/viewsets/model.py#L515-L517

The generated menu links are titlecased, which is very "english", is there any reason for using `.title()` as these labels are already translated and user-facing strings?

**Comment by gasman** [comment]
The convention [as suggested by Django](https://docs.djangoproject.com/en/5.1/topics/db/models/#verbose-field-names) is for `verbose_name` / `verbose_name_plural` to be left as lower-case (which allows it to be used in contexts like "There are no matching {verbose_name_plural}."), so the `title()` is necessary to capitalise it in the way that you'd expect it to appear as a menu label - in English at least. I wasn't aware that this was particularly specific to English though - can you give any examples of languages where this wouldn't be the desired behaviour?

In any case, this is only a default for when `menu_label` isn't explicitly defined - if you're working with a name where normal capitalisation rules don't apply, you can define something like `menu_label = _("e-Scooters")` and ensure that your translations for that string use the correct capitalisation.

**Reply by SebCorbin** [reply]
I was not aware of the convention, but as you mentionned, the menu labels should be *capitalized* and not *titlecased*, hence resulting (for example with `verbose_name_plural=_("mapping interfaces")`) in "Mapping interfaces" instead of "Mapping Interfaces"

But with another example (let's take "IP addresses"), we should have neither "Ip adresses" (capitalized) or "Ip Addresses" (titlecases) but "IP addresses"

Or with "snippets from Django", we should have neither "Snippets from django" (capitalized) or "Snippets From Django" but "Snippets from Django"

**Reply by laymonage** [reply]
That being said, the menu items in "Reports" and "Settings" are sentence-cased rather than title-cased, so perhaps the viewset's default should use `capfirst()` instead of `.title()`.

The only reason I can think of why this uses `.title()` was to follow ModelAdmin: https://github.com/wagtail-nest/wagtail-modeladmin/blob/742fbe8d5e942ba641af4c25fb181c65ad4858ed/wagtail_modeladmin/options.py#L208

**Reply by gasman** [reply]
Ah yep, this has come up before (e.g. in #9219 and #10434) and we decided then that we should standardise on Django's `capfirst`.

**Reply by SebCorbin** [reply]
Great, I'll open a pull request then (edit: https://github.com/wagtail/wagtail/pull/12358)
